modules/campaignion_conf/campaignion_conf.module

```diff
@@ -12,6 +12,14 @@ function campaignion_conf_module_implements_alter(&$implementations, $hook) {
     unset($implementations['image']);
     unset($implementations['media_youtube']);
   }
+  if ($hook == 'features_pipe_node_alter') {
+    // Ignore diff & comment variables for node-types.
```

Comment:
Is this comment in the right place? I find it rather confusing TBH.

A comment about filtering comment variables in `campaignion_conf_features_pipe_node_alter` on the other hand would be nice.

Reply:
Yes it’s in the right place. Removing the diff-hook is all that needs to be done for removing the diff variables.

Reply:
Comments is more complicated as it is added by strongarm together with other variables (that we do need). So the purpose of the whole section in between the curly-praces is exactly what’s stated. That‘s the “why?” for this block of code.

Reply:
But the comment is *inside* the curly brackets so it just comments `unset($implementations['diff']);` which has nothing to do with comment variables at all and should only say "Ignore diff variables for node-types.". The filtering comment variables part is actually in the hook that's just being moved to being called as late as possible here.

Reply:
I’ve moved the comment one line up.

Reply:
What about this? More consistency, less duplicate code:

```php
/**
 * Implements hook_module_implements_alter().
 */
function campaignion_conf_module_implements_alter(&$implementations, $hook) {
  if ($hook == 'file_default_displays_alter') {
    unset($implementations['image']);
    unset($implementations['media_youtube']);
  }
  // Ignore diff variables for node-types.
  if ($hook == 'features_pipe_node_alter') {
    unset($implementations['diff']);
  }
  // Ensure our hook implementations are called late.
  if (in_array($hook, ['features_pipe_node_alter', 'features_pipe_context_alter'])) {
    // Ensure our hook implementation is called late.
    $group = $implementations['campaignion_conf'];
    unset($implementations['campaignion_conf']);
    $implementations['campaignion_conf'] = $group;
  }
}

/**
 * Implements hook_features_pipe_COMPONENT_alter() for context.
 */
function campaignion_conf_features_pipe_context_alter(&$pipe, $data, $export) {
  // Deactivate content-type dependencies caused by context conditions.
  unset($pipe['node']);
}

/**
 * Implements hook_features_pipe_COMPONENT_alter() for node.
 */
function campaignion_conf_features_pipe_node_alter(&$pipe, $data, $export) {
  // Ignore comment variables for node-types.
  $pipe['variable'] = array_filter($pipe['variable'], function ($name) {
    if (substr($name, 0, 8) == 'comment_') {
      return FALSE;
    }
    return TRUE;
  });
}
```

Reply:
Ok, will change it accordingly …

Reply:
Or not :thinking: The reason for this that I usually want to know what happens to each hook. I can see that more clearly with the previous version, although it has more code duplication.

Reply:
How about factoring it out like this?

Reply:
Ok, then:

```php
/**
 * Implements hook_module_implements_alter().
 */
function campaignion_conf_module_implements_alter(&$implementations, $hook) {
  if ($hook == 'file_default_displays_alter') {
    unset($implementations['image']);
    unset($implementations['media_youtube']);
  }
  if ($hook == 'features_pipe_context_alter') {
    // Ensure our hook implementations are called late.
    $group = $implementations['campaignion_conf'];
    unset($implementations['campaignion_conf']);
    $implementations['campaignion_conf'] = $group;
  }
  if ($hook == 'features_pipe_node_alter') {
    // Ignore diff variables for node-types.
    unset($implementations['diff']);
    // Ensure our hook implementations are called late.
    $group = $implementations['campaignion_conf'];
    unset($implementations['campaignion_conf']);
    $implementations['campaignion_conf'] = $group;
  }
}

/**
 * Implements hook_features_pipe_COMPONENT_alter() for context.
 */
function campaignion_conf_features_pipe_context_alter(&$pipe, $data, $export) {
  // Deactivate content-type dependencies caused by context conditions.
  unset($pipe['node']);
}

/**
 * Implements hook_features_pipe_COMPONENT_alter() for node.
 */
function campaignion_conf_features_pipe_node_alter(&$pipe, $data, $export) {
  // Ignore comment variables for node-types.
  $pipe['variable'] = array_filter($pipe['variable'], function ($name) {
    if (substr($name, 0, 8) == 'comment_') {
      return FALSE;
    }
    return TRUE;
  });
}
```

Reply:
:+1: for factoring out, still not happy about comments :wink:

Reply:
Hm, now the comment before `$move_to_the_end` is lying unless I hardcode `$module='campaignion_conf'`. Can I just remove the comment?

Reply:
Why, wouldn't it be called late for any module? :thinking: But fine with not commenting the function.

Could you also fix the indentation and maybe add an overall function comment? Thanks!

Reply:
… and also fixed the wrong intentation. If not I’d suggest use some real-time code collaboration for this …

Reply:
Looks perfect now! 

Except for the missing doc comment but that's your domain usually…